amethyst_renderer/src/scene.rs

```diff
@@ -0,0 +1,63 @@
+//! A fully renderable scene.
+//!
+//! # Example
+//!
+//! ```rust
+//! let mut scene = Scene::default();
+//! scene.add_light(PointLight::default());
+//! scene.add_mesh(sphere, material);
+//! ```
+
+use cam::Camera;
+use fnv::FnvHashMap as HashMap;
+use light::Light;
+use mesh::Mesh;
+use mtl::Material;
+use std::collections::hash_map::Values;
+
+/// Immutable slice iterator of lights.
+pub type LightIter<'l> = Values<'l, String, Light>;
+
+/// Immutable slice iterator of meshes.
+pub type MeshIter<'m> = Values<'m, String, Mesh>;
+
+/// Collection of lights and meshes to render.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Scene {
+    camera: Vec<Camera>,
+    lights: HashMap<String, Light>,
```

Comment:
`Scene` is read every frame and there might be a lot of lights, materials, meshes. So maybe use `Vec<Thing>` here? Because `HashMap<String, Thing>` is expensive (AFAIK it is less cache friendly than `Vec`, because elements are scattered around memory - so `lights.values()` and `light.values().next()` aren't guaranteed to come one after another in RAM, hence it is less likely that both `lights.values()` and `light.values().next()` would be in cache when read).